### PR TITLE
feat(arc-saas):Integrate backend with arc-saas-ui

### PR DESCRIPTION
**Pull request by Deepika516**
## Description

frontend backend Integration

Fixes #34 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yeshamavani** [comment]
@Surbhi-sharma1 @Deepika516 why is commitlint not configured here 
this is not how we write our commit messages 
raise a separate PR for that first

File: projects/saas-ui/src/app/main/plan-items/plan-items.component.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    {field: 'feature2', editable: true},
+  ];
+
+  rowData = [
+    {
+      featureName: 'S2 service',
+      featureDescription: 'AWS Service',
+      feature1: 'EC2',
+      feature2: 'ECS',
+    },
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/plan-items/plan-items.component.ts:45]: ??

File: projects/saas-ui/src/app/main/plan-items/plan-items.component.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  planItems() {
+    // this.tenantFacade
+    //   .getTenantList()
+    //   .pipe(takeUntil(this._destroy$))
+    //   .subscribe(res => {
+    //     this.tenants = res;
+    //   });
+  }
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/plan-items/plan-items.component.ts:65]: ??

File: projects/saas-ui/src/app/main/plan-items/plan-items.component.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  onAddButtonClick() {
+    const newItem = {
+      featureName: 'New feature', // Provide default values or make them empty strings
+      featureDescription: 'Description',
+      feature1: 'Detail 1',
+      feature2: 'Detail 2',
+    };
+    this.rowData.push(newItem);
+    this.gridApi.setRowData(this.rowData); // Update the grid data
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/plan-items/plan-items.component.ts:88]: hard code

File: projects/saas-ui/src/app/main/tenant-list/commands/get-tenant-lead.command.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    apiService: ApiService,
+    adapter: IAdapter<T>,
+    @Inject(APP_CONFIG) private readonly appConfig?: IAnyObject,
+  ) {
+    super(
+      apiService,
+      adapter,
+      `${appConfig.baseApiUrl}${appConfig.tenantMgmtFacadeUrl}/tenants`, // if error remove ${appConfig.baseApiUrl}
+    );
+  }
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/tenant-list/commands/get-tenant-lead.command.ts:19]: why this comment 
is the baseApiUrl needed or not ?

File: projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+@Injectable({
+  providedIn: ApiModule,
+})
+export class ValidateLeadAdapter implements IAdapter<any> {
+  adaptToModel(resp: any): any {
+    return resp.body;
+  }
+  adaptFromModel(data: any): any {
+    return data;
+  }
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/adapters/validate-lead-adapter.service.ts:14]: If there is nothing specific and we are using any types only 
then why not use AnyAdapter

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.html [hunk 6]
```diff
@@ -0,0 +1,185 @@
+<nb-card class="h-100 card-row">
+  <nb-card-body class="m-0">
+    <div class="main-wrapper add-lead-wrapper">
+      <div>
+        <div class="sign-in-title">
+          <h2>Add Lead</h2>
+          <h3>Welcome to SAAS UI by SourceFuse</h3>
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.html:7]: do the changes we discussed on call

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      state: ['', Validators.required],
+      city: ['', Validators.required],
+      zip: ['', Validators.required],
+    });
+  }
+
+  onSubmit() {
+    if (this.addLeadForm.valid) {
+      const userData = this.addLeadForm.value;
+      console.log(userData);
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/add-lead/add-lead.component.ts:40]: remove console

File: projects/saas-ui/src/app/on-boarding/add-lead/add-lead.module.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import {NgModule} from '@angular/core';
+import {CommonModule} from '@angular/common';
+
+import {AddLeadRoutingModule} from './add-lead-routing.module';
+
+@NgModule({
+  declarations: [],
+  imports: [CommonModule, AddLeadRoutingModule],
+})
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/add-lead/add-lead.module.ts:9]: why make modules for each component 
when they can be managed under same root module of on-boarding

File: projects/saas-ui/src/app/on-boarding/add-tenant/add-tenant.component.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      price: '$30/month',
+      features: [
+        'Feature 1',
+        'Feature 2',
+        'Feature 3',
+        'Feature 4',
+        'Feature 5',
+      ],
+    },
+  ];
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/add-tenant/add-tenant.component.ts:58]: it should be fetch via api call

File: projects/saas-ui/src/app/on-boarding/commands/delete-lead-command.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/commands/delete-lead-command.ts:1]: where is this being used ?

File: projects/saas-ui/src/app/on-boarding/commands/get-lead-command.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    apiService: ApiService,
+    adapter: IAdapter<Lead>,
+    @Inject(APP_CONFIG) private readonly appConfig?: IAnyObject,
+  ) {
+    super(
+      apiService,
+      adapter,
+      `${appConfig.baseApiUrl}/${appConfig.userServiceUrl}/leads`,
+    );
+  }
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/commands/get-lead-command.ts:23]: somewhere baseurl is included and some places its not
why this ambiguity

File: projects/saas-ui/src/app/on-boarding/commands/get-lead-by-id.command.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    apiService: IApiService,
+    adapter: IAdapter<Lead>,
+    leadId: string,
+    @Inject(APP_CONFIG) private readonly appConfig?: IAnyObject,
+  ) {
+    super(
+      apiService,
+      adapter,
+      `${appConfig.baseApiUrl}${appConfig.tenantmgmtServiceUrl}/leads/${leadId}`,
+    );
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/commands/get-lead-by-id.command.ts:25]: someplaces its tenant mgmt service while some place is facade 
why create confusion for user

File: projects/saas-ui/src/app/on-boarding/email-verification/email-verification.component.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(
+    private route: ActivatedRoute,
+    private location: Location,
+    private readonly authService: AuthService,
+    private readonly router: Router,
+    private fb: FormBuilder,
+  ) {}
+
+  completeRegistration() {
+    this.router.navigate(['/tenant/add-tenant']);
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/email-verification/email-verification.component.ts:23]: why a redirection from here 
add-tenant will be called through the link send via email

File: projects/saas-ui/src/app/on-boarding/on-boarding-routing.module.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+      },
+      {
+        path: 'registration/complete',
+        component: VerificationCompleteComponent,
+      },
+      {
+        path: 'add-tenant',
+        // path: 'verify/:leadId',
+        component: AddTenantComponent,
+        // canActivate: [EmailVerifyGuard],
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/on-boarding-routing.module.ts:30]: ??

File: projects/saas-ui/src/app/on-boarding/on-boarding-service.ts [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+    leadId: string,
+  ): Observable<{leadId: string; token: string}> {
+    const command: VerifyEmailCommand<{leadId: string; token: string}> =
+      new VerifyEmailCommand(this.apiService, this.validateLeadAdapter, leadId);
+    // sonarignore:end
+    command.parameters = {
+      data: {},
+      observe: 'response',
+      headers: new HttpHeaders()
+        .set(AuthTokenSkipHeader, '-')
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/on-boarding/on-boarding-service.ts:53]: why skip header here

File: projects/saas-ui/src/app/shared /api/api.service.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /api/api.service.ts:1]: has a space in name

File: projects/saas-ui/src/app/shared /auth/auth.service.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/auth.service.ts:1]: why repeating the auth service in lib and project

File: projects/saas-ui/src/app/shared /auth/auth.service.ts [hunk 15]
```diff
@@ -0,0 +1,433 @@
...
+      return command.execute().pipe(
+        tap(res => {
+          this.store.setUser(res);
+          this._loadPermissions(res.permissions);
+        }),
+      );
+    }
+  }
+
+  // sonarignore:start
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/auth.service.ts:102]: why sonarignore

File: projects/saas-ui/src/app/shared /auth/auth.service.ts [hunk 16]
```diff
@@ -0,0 +1,433 @@
...
+      new VerifyResetPasswordLinkCommand(
+        this.apiService,
+        this.anyAdapter,
+        this.appConfig,
+      );
+    // sonarignore:end
+    command.parameters = {
+      data: {
+        token: token,
+        client_id: this.appConfig.clientId,
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/auth.service.ts:135]: where are these function being used if not needed please remove this

File: projects/saas-ui/src/app/shared /auth/auth.service.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/auth.service.ts:1]: multiple functions in this file that are most probably not being used anywhere 
remove them keep only required code

File: projects/saas-ui/src/app/shared /auth/auth.service.ts [hunk 17]
```diff
@@ -0,0 +1,433 @@
...
+        client_secret: this.appConfig.publicKey,
+      },
+      observe: 'response',
+      headers: this.authTokenSkipHeader,
+    };
+    return command.execute();
+  }
+
+  loginViaGoogle(): void {
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/auth.service.ts:191]: this is not being used

File: projects/saas-ui/src/app/shared /auth/commands/del-api.command.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/commands/del-api.command.ts:1]: this shoudl have been a part of the library, dont repeat the code 
same for all base commads

File: projects/saas-ui/src/app/shared /auth/commands/google-login.command.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/shared /auth/commands/google-login.command.ts:1]: again

File: projects/saas-ui/src/assets/images/Illustration.svg
Comment (yeshamavani) [inline projects/saas-ui/src/assets/images/Illustration.svg:1]: where is this image used

Comment (yeshamavani) [reply projects/saas-ui/src/assets/images/Illustration.svg:1]: also why are we repeating images as well 
should not this also be a part of the library

File: projects/saas-ui/src/assets/images/auth/logo.png
Comment (yeshamavani) [inline projects/saas-ui/src/assets/images/auth/logo.png:1]: ??

**Review by yeshamavani — CHANGES_REQUESTED** [review]

**Comment by Surbhi-sharma1** [comment]
Closed as not required

**Comment by Surbhi-sharma1** [comment]
Close

**Pull request closed without merging**